src/scenario_tests.rs

```diff
@@ -876,7 +887,73 @@ mod dst_tests {
                 ParsecVote::CheckElder.to_event(),
             ],
             &AssertState {
-                action_our_events: vec![Rpc::Merge.to_event()],
+                action_our_events: vec![Rpc::Merge(SectionInfo(OUR_SECTION, 0)).to_event()],
+            },
+        );
+    }
+
+    #[test]
+    fn parsec_merge_sibling() {
+        let initial_state = arrange_initial_state(
+            &initial_state_old_elders(),
+            &[
+                TestEvent::SetChurnNeeded(ChurnNeeded::Merge).to_event(),
+                ParsecVote::CheckElder.to_event(),
+            ],
+        );
+        run_test(
+            "Decide to merge, and then later store merge infos in ProcessMerge",
+            &initial_state,
+            &[ParsecVote::NeighbourMerge(OTHER_SECTION_INFO).to_event()],
+            &AssertState {
+                action_our_events: vec![
+                    ActionTriggered::MergeInfoStored(OTHER_SECTION_INFO).to_event(),
+                    ParsecVote::NewSectionInfo(MERGED_SECTION_INFO).to_event(),
+                ],
+            },
+        );
+    }
+
+    #[test]
+    fn parsec_merge_non_sibling() {
+        let initial_state = arrange_initial_state(
+            &initial_state_old_elders(),
+            &[
+                TestEvent::SetChurnNeeded(ChurnNeeded::Merge).to_event(),
+                ParsecVote::CheckElder.to_event(),
+            ],
+        );
+        run_test(
+            "Get consensus on merging with a non-sibling neighbour",
+            &initial_state,
+            &[ParsecVote::NeighbourMerge(REMOTE_OTHER_SECTION_INFO).to_event()],
+            &AssertState {
+                action_our_events: vec![ActionTriggered::MergeInfoStored(
+                    REMOTE_OTHER_SECTION_INFO,
+                )
+                .to_event()],
+            },
+        );
+    }
+
+    #[test]
+    fn parsec_new_section() {
```

Comment:
maybe call it `merge_complete` or something like it.

Reply:
Without the `parsec_` prefix?

Reply:
Actually, ignore that, your name is consistent with the others.